### PR TITLE
packetbeat/config: allow multiple interfaces to be specified in the config

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

Currently only the first interface is used, retaining the existing behaviour. If
command line arguments are provided the existing behaviour is kept with the first
interface having options from the command line replacing being placed in that
config as defaults, replaced by the options in the config file.

## Why is it important?

This is required for allowing multiple interfaces to be specified.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] ~~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~ **Not strictly user-facing.**

## Author's Checklist

- [ ]

## Related issues

- For #31905

### CODE REVIEW

**Comment by efd6** [comment]
The only change between the passing state and the current failing in the packetbeat directory is a change in the dockerfile's Go version
```
$ git diff a1ab202137 cc24601258 -- .
diff --git a/packetbeat/Dockerfile b/packetbeat/Dockerfile
index 9ba19cbbc6..d24a21f653 100644
--- a/packetbeat/Dockerfile
+++ b/packetbeat/Dockerfile
@@ -1,4 +1,4 @@
-FROM golang:1.18.6
+FROM golang:1.18.5
 
 RUN \
     apt-get update \
```

**Comment by efd6** [comment]
The failures must have been due to more rigorous testing, they found real failures that were missed previously :tada:

File: packetbeat/config/config.go [hunk 1]
```diff
@@ -28,7 +31,8 @@ import (
 )
 
 type Config struct {
-	Interfaces      InterfacesConfig   `config:"interfaces"`
+	Interface       *InterfacesConfig  `config:"interfaces"`
+	Interfaces      []InterfacesConfig `config:"interfaces"`
```
[the remark this answers is not in the document]
Comment (efd6) [reply packetbeat/config/config.go:35]: I'd be inclined to do this in a follow-up PR.

**Review by andrewkroh — APPROVED** [review]
I'm good with having the comments addressed as a follow-up.

**Pull request merged**